src/test/java/org/opensearch/neuralsearch/plugin/query/NeuralQueryIT.java

```diff
@@ -0,0 +1,294 @@
+     * {
+     *     "query": {
+     *         "neural": {
+     *             "text_knn": {
+     *                 "query_text": "Hello world",
+     *                 "model_id": "dcsdcasd",
+     *                 "k": 1
+     *             }
+     *         }
+     *     }
+     * }
+     */
+    @SneakyThrows
+    public void testBasicQuery() {
+        initializeIndexIfNotExist(TEST_BASIC_INDEX_NAME);
+        NeuralQueryBuilder neuralQueryBuilder = new NeuralQueryBuilder(
+            TEST_KNN_VECTOR_FIELD_NAME_1,
+            TEST_QUERY_TEXT,
+            modelId.get(),
+            1,
+            null
+        );
+        Map<String, Object> searchResponseAsMap = search(TEST_BASIC_INDEX_NAME, neuralQueryBuilder, 1);
+        Map<String, Object> firstInnerHit = getFirstInnerHit(searchResponseAsMap);
+
+        assertEquals("1", firstInnerHit.get("_id"));
+        float expectedScore = computeExpectedScore(modelId.get(), testVector, TEST_SPACE_TYPE, TEST_QUERY_TEXT);
+        assertEquals(expectedScore, objectToFloat(firstInnerHit.get("_score")), 0.0);
+    }
+
+    /**
+     * Tests basic query with boost parameter:
+     * {
+     *     "query": {
+     *         "neural": {
+     *             "text_knn": {
+     *                 "query_text": "Hello world",
+     *                 "model_id": "dcsdcasd",
+     *                 "k": 1,
+     *                 "boost": 2.0
+     *             }
+     *         }
+     *     }
+     * }
+     */
+    @SneakyThrows
+    public void testBoostQuery() {
+        initializeIndexIfNotExist(TEST_BASIC_INDEX_NAME);
+        NeuralQueryBuilder neuralQueryBuilder = new NeuralQueryBuilder(
+            TEST_KNN_VECTOR_FIELD_NAME_1,
+            TEST_QUERY_TEXT,
+            modelId.get(),
+            1,
+            null
+        );
+
+        final float boost = 2.0f;
+        neuralQueryBuilder.boost(boost);
+        Map<String, Object> searchResponseAsMap = search(TEST_BASIC_INDEX_NAME, neuralQueryBuilder, 1);
+        Map<String, Object> firstInnerHit = getFirstInnerHit(searchResponseAsMap);
+
+        assertEquals("1", firstInnerHit.get("_id"));
+        float expectedScore = 2 * computeExpectedScore(modelId.get(), testVector, TEST_SPACE_TYPE, TEST_QUERY_TEXT);
+        assertEquals(expectedScore, objectToFloat(firstInnerHit.get("_score")), 0.0);
+    }
+
+    /**
+     * Tests rescore query:
+     * {
+     *     "query" : {
+     *       "match_all": {}
+     *     },
+     *     "rescore": {
+     *         "query": {
+     *              "rescore_query": {
+     *                  "neural": {
+     *                      "text_knn": {
+     *                          "query_text": "Hello world",
+     *                          "model_id": "dcsdcasd",
+     *                          "k": 1
+     *                      }
+     *                  }
+     *              }
+     *          }
+     *    }
+     */
+    @SneakyThrows
+    public void testRescoreQuery() {
```

Comment:
Should we add some function to rescore like weight?

Reply:
I didnt think it was necessary given that weight will be outside of the neural query